File: android/app/src/main/java/com/guitaripod/pixie/data/api/model/AdminModels.kt
package com.guitaripod.pixie.data.api.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class SystemStatsResponse(
    val users: UserStats,
    val credits: CreditStats,
    val revenue: RevenueStats,
    val images: ImageStats
)

@JsonClass(generateAdapter = true)
data class UserStats(
    val total: Int
)

@JsonClass(generateAdapter = true)
data class CreditStats(
    @Json(name = "total_balance") val totalBalance: Int,
    @Json(name = "total_purchased") val totalPurchased: Int,
    @Json(name = "total_spent") val totalSpent: Int
)

@JsonClass(generateAdapter = true)
data class RevenueStats(
    @Json(name = "total_usd") val totalUsd: String,
    @Json(name = "openai_costs_usd") val openaiCostsUsd: String,
    @Json(name = "gross_profit_usd") val grossProfitUsd: String,
    @Json(name = "profit_margin") val profitMargin: String
)

@JsonClass(generateAdapter = true)
data class ImageStats(
    @Json(name = "total_generated") val totalGenerated: Int
)

@JsonClass(generateAdapter = true)
data class AdminCreditAdjustmentRequest(
    @Json(name = "user_id") val userId: String,
    val amount: Int,
    val reason: String
)

@JsonClass(generateAdapter = true)
data class AdminCreditAdjustmentResponse(
    @Json(name = "new_balance") val newBalance: Int
)

@JsonClass(generateAdapter = true)
data class UserSearchResult(
    val id: String,
    val email: String?,
    @Json(name = "is_admin") val isAdmin: Boolean,
    val credits: Int,
    @Json(name = "created_at") val createdAt: String
)

@JsonClass(generateAdapter = true)
data class AdjustmentHistoryItem(
    val id: String,
    @Json(name = "user_id") val userId: String,
    @Json(name = "admin_id") val adminId: String,
    val amount: Int,
    val reason: String,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "new_balance") val newBalance: Int
)

@JsonClass(generateAdapter = true)
data class AdjustmentHistoryResponse(
    val adjustments: List<AdjustmentHistoryItem>
)

File: iOS/Pixie/Pixie/Data/Models/AdminModels.swift
import Foundation

struct SystemStatsResponse: Codable {
    let users: UserStats
    let credits: CreditStats
    let revenue: RevenueStats
    let images: ImageStats
}

struct UserStats: Codable {
    let total: Int
}

struct CreditStats: Codable {
    let totalBalance: Int
    let totalPurchased: Int
    let totalSpent: Int
    
    enum CodingKeys: String, CodingKey {
        case totalBalance = "total_balance"
        case totalPurchased = "total_purchased"
        case totalSpent = "total_spent"
    }
}

struct RevenueStats: Codable {
    let totalUsd: String
    let openaiCostsUsd: String
    let grossProfitUsd: String
    let profitMargin: String
    
    enum CodingKeys: String, CodingKey {
        case totalUsd = "total_usd"
        case openaiCostsUsd = "openai_costs_usd"
        case grossProfitUsd = "gross_profit_usd"
        case profitMargin = "profit_margin"
    }
}

struct ImageStats: Codable {
    let totalGenerated: Int
    
    enum CodingKeys: String, CodingKey {
        case totalGenerated = "total_generated"
    }
}

struct AdminCreditAdjustmentRequest: Codable {
    let userId: String
    let amount: Int
    let reason: String
    
    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case amount, reason
    }
}

struct AdminCreditAdjustmentResponse: Codable {
    let newBalance: Int
    
    enum CodingKeys: String, CodingKey {
        case newBalance = "new_balance"
    }
}

struct UserSearchResult: Codable {
    let id: String
    let email: String?
    let isAdmin: Bool
    let credits: Int
    let createdAt: String
    
    enum CodingKeys: String, CodingKey {
        case id, email, credits
        case isAdmin = "is_admin"
        case createdAt = "created_at"
    }
}

struct AdjustmentHistoryItem: Codable {
    let id: String
    let userId: String
    let adminId: String
    let amount: Int
    let reason: String
    let createdAt: String
    let newBalance: Int
    
    enum CodingKeys: String, CodingKey {
        case id, amount, reason
        case userId = "user_id"
        case adminId = "admin_id"
        case createdAt = "created_at"
        case newBalance = "new_balance"
    }
}

struct AdjustmentHistoryResponse: Codable {
    let adjustments: [AdjustmentHistoryItem]
}